import "kademlia_service_messages.proto";

package kad;

service KademliaService {
  rpc Ping (PingRequest) returns (PingResponse);
  rpc FindValue (FindRequest) returns (FindResponse);
  rpc FindNode (FindRequest) returns (FindResponse);
  rpc Store (StoreRequest) returns (StoreResponse);
  rpc Downlist (DownlistRequest) returns (DownlistResponse);
  rpc NatDetection (NatDetectionRequest) returns (NatDetectionResponse);
  rpc NatDetectionPing (NatDetectionPingRequest) returns (NatDetectionPingResponse);
  rpc Bootstrap (BootstrapRequest) returns (BootstrapResponse);
}
